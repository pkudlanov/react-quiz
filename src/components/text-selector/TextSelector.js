import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({
  text,
  handleTextChange,
  handleSubmit
}){
  return(
    <form onSubmit={handleSubmit} >
      <input
        type="text"
        value={text}
        onChange={handleTextChange} />
      <button>Submit</button>
    </form>
  );
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  handleTextChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default TextSelector;
