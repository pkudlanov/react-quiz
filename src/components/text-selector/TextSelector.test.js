import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('renders TextSelector', () => {
    const wrapper = shallow(<TextSelector
      text="Greetings friends and foe!!!!"
      handleTextChange={() => { }}
      handleSubmit={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
