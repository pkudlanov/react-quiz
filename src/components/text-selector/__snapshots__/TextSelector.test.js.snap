// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders TextSelector 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="Greetings friends and foe!!!!"
  />
  <button>
    Submit
  </button>
</form>
`;
